src/common/imageio_gm.c

```diff
@@ -119,6 +119,15 @@ dt_imageio_retval_t dt_imageio_open_gm(dt_image_t *img, const char *filename, dt
     }
   }
 
+  size_t profile_length;
```

Comment:
I _think_ GM already converts pixels to sRGB internally, so there is perhaps no point in getting a profile here?

Reply:
And I guess I was wrong! It looks like there is no conversion to sRGB if the GM API is used (unlike the CLI tools), and the profile is indeed useful, and seems to work (i.e. switching between the detected "embedded ICC" and "HLG Rec 2020" makes no difference, as expected; I'll attach my test image to the issue rather than here).